### DISCUSSION TITLE
How to fit the textbox width in group when scaling ?

### DISCUSSION THREAD

**Discussion by senschen** [fabricjs/fabric.js#9122] [Q&A] [answered]
I want to not scale the textbox when scaling , and only change its width,
I finish it for textbox , but it seems not a perfect solution.

```
canvas.on('object:scaling', (e) => {
  if (e.target.type === 'textbox') {
    e.target.width *= e.target.scaleX;
    // e.target.height *= e.target.scaleY;
    e.target.scaleX = 1;
    e.target.scaleY = 1;
    e.target.setCoords();
  } else if (e.target.type === 'group') {
    const group = e.target;
    const rect = group.getObjects()[0];
    const text = group.getObjects()[1];
    text.scaleX = 1 / group.scaleX;
    text.scaleY = 1 / group.scaleY;
    text.width = rect.width * group.scaleX;
    text.setCoords();
 }
});
```

this is my demo link: 
https://codepen.io/senschen/pen/WNYErBK

https://github.com/fabricjs/fabric.js/assets/8544826/466960ce-42e3-4395-9f3c-1fefa99e1628

I want to improve the solution with createClass or change the code logic when render text with scaleX/Y, and not to change the scaleX to 1.
could you give some ideas for it ?  thank you very much.

**Comment by ShaMan123** [comment]
Textbox controls are already supported, don't waste time on workarounds

**Reply by ShaMan123** [reply, marked as answer]
Look at the source code and adjust yuor controls accordingly 
I remember I wrote a PR for this but can't find it
https://github.com/fabricjs/fabric.js/blob/master/src/controls/changeWidth.ts

**Comment by senschen** [comment]
> Look at the source code and adjust yuor controls accordingly I remember I wrote a PR for this but can't find it https://github.com/fabricjs/fabric.js/blob/master/src/controls/changeWidth.ts

But it seems only works for text, not adapt group(rect + text), it may also be that I didn't understand your suggestion.

I want this result: 
when scaling the group by bottom-right-control (uniformScaling = false), the text could autolly resizing width according to its group's scaleX value, **just like other strokeUniform shapes in group.**

**Reply by ShaMan123** [reply]
then you need to adapt group controls as well or listen to the proper event

**Reply by senschen** [reply]
sorry I can not understant how to do. do you mean I need to change the controls.br actionHandler to achive it like this ?
```
actionHandler: () => { 
    scalingEqually();   // for shapes in group
    changeWidth();    // for textbox in group
 }
```

**Reply by ShaMan123** [reply]
Yes, something like that
